### DISCUSSION TITLE
Browser Crash Issue When Loading High-Resolution Images in OpenSeaDragon

### DISCUSSION THREAD

**Discussion by harshkumarlaldas** [openseadragon/openseadragon#2417] [Q&A]
The browser is experiencing crashes when attempting to load high-resolution images in OpenSeaDragon from AWS s3, especially when zoomed in or moved. This issue is primarily due to the combination of large tile sizes (>7MB) and a significant number of concurrent network requests (around >10) for each image. The extended time required for downloading each image (approximately 30-90 seconds) eventually overloads the browser, leading to crashes and rendering a black screen across the browser.

**Issue Details:**

Tile Size: The issue is related to the size of individual image tiles, with each tile exceeding 7MB in size. These large tiles are commonly used to display high-resolution images.

Network Calls: For each image being displayed in the OpenSeaDragon viewer, there are more than 10 network calls simultaneously initiated. These calls are responsible for fetching the image tiles.

Download Time: Due to the combination of large tile sizes and multiple concurrent network requests, it takes a considerable amount of time (ranging from 30 to 90 seconds) for each image to complete downloading.

Browser Crashes: The extended loading times and the substantial amount of data being processed eventually cause the browser to become unresponsive and crash. Users are left with a black screen in their browser window.

Url Generation: A preSigned Url is being attached to the getTileUrl function, which gets called once the OSD dom sets in.
This Url connects to a bucket in AWS s3, which leads to the specific tiles to get downloaded.

**Steps to Reproduce:**

Launch the application with OpenSeaDragon integration.
Navigate to a high-resolution image.
Zoom in or move within the image to trigger tile loading.
Observe the extended loading times and the eventual browser crash.

**Expected Behavior:**

The browser should be able to handle the loading of high-resolution images in OpenSeaDragon without crashing or causing a black screen.

I did upgrade to the latest version of the OSD and the issue still exists.

Any input will be immensely helpful.

**Comment by pearcetm** [comment]
If you can break your large tiles into multiple smaller ones forming an image pyramid, that should work much better since that's what OSD is intended to work with. Do you need pointers on how to go about that?
